sklearn/utils/weight_vector.pyx

```diff
@@ -60,7 +58,7 @@ cdef class WeightVector(object):
 
         self.aw = aw
         if self.aw is not None:
-            self.aw_data_ptr = <double *>aw.data
+            self.aw_data_ptr = &aw[0]
```

Comment:
Would this work?

```suggestion
            self.aw_data_ptr = &aw
```

Also maybe we don't even need the pointer anymore?

Reply:
We still need the `[0]` here: http://docs.cython.org/en/latest/src/userguide/memoryviews.html#pass-data-from-a-c-function-via-pointer

Reply:
This pointer may not be needed.

Reply:
Kind of weird since `WeightVector` is public.